### DISCUSSION TITLE
When to chose to add new OSD rather to expand an existing OSD volume

### DISCUSSION THREAD

**Discussion by Omar-Challouf** [rook/rook#8499] [Q&A] [answered]
Hello,

Could you please explain the use case to add a new OSD to the cluster when you have the possibility to expand an existing OSD volume?

Thank you,

**Comment by travisn** [comment, marked as answer]
The resources required to run the OSDs will be affected by either expanding existing OSDs or adding new OSDs. While there is no exact answer, here are some considerations:
- If the OSD is CPU-bound, you would want to add more OSDs instead of expanding the existing OSDs. This will usually only happen for fast devices such as NVMe, but not hdds.
- If the OSD is very large (e.g. 4TiB) you likely should add more OSDs instead of continuing to expand.
- Data loss is most commonly because of individual device failure, which causes failure of an entire OSD. The bigger an OSD, the more data will be lost if the OSD fails. Ceph would recover the data (PGs) since there are multiple replicas, but it will take longer with more lost data.

**Reply by Omar-Challouf** [reply]
Thank you, sir.
